### ISSUE TITLE
Cannot install oci8 on PHP 8.2

### ISSUE THREAD

**Issue by chieroz** [laradock/laradock#3532]
### Description:

I tried to install oci8 on PHP 8.2 but I get an error message about the Oracle Instant Client.

### Expected Behavior:

The installation should complete successfully

### Context information:

**Output of `git rev-parse HEAD`**

```
505130302703a85415cfb011ed320eddd88c92e4
```

**Output of `docker version`**

```
Client:
 Version:           26.1.4
 API version:       1.45
 Go version:        go1.21.11
 Git commit:        5650f9b
 Built:             Wed Jun  5 11:26:02 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.31.0 (153195)
 Engine:
  Version:          26.1.4
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.11
  Git commit:       de5c9cf
  Built:            Wed Jun  5 11:29:12 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.33
  GitCommit:        d2d58213f83a351ca8f528a95fbd145f5654e957
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker-compose version`**

```
Docker Compose version v2.27.1-desktop.1
```

**System info: Mac, Windows or Linux. Include which disto/version**

```
MacBook Pro with Apple M1 Pro, OS Mac Sonoma 14.5
```

### Steps to reproduce the issue:

1. set PHP_VERSION=8.2 in .env
2. set PHP_FPM_INSTALL_OCI8=true in .env
3. launch 'docker-compose up -d nginx mysql'

### Stacktrace & Additional info:

```
30.81 running: make
30.82 /bin/bash /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/libtool --tag=CC --mode=compile cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8.c -o oci8.lo  -MMD -MF oci8.dep -MT oci8.lo
30.83 libtool: compile:  cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8.c -MMD -MF oci8.dep -MT oci8.lo  -fPIC -DPIC -o .libs/oci8.o
31.33 /bin/bash /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/libtool --tag=CC --mode=compile cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_lob.c -o oci8_lob.lo  -MMD -MF oci8_lob.dep -MT oci8_lob.lo
31.35 libtool: compile:  cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_lob.c -MMD -MF oci8_lob.dep -MT oci8_lob.lo  -fPIC -DPIC -o .libs/oci8_lob.o
31.61 /bin/bash /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/libtool --tag=CC --mode=compile cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_statement.c -o oci8_statement.lo  -MMD -MF oci8_statement.dep -MT oci8_statement.lo
31.63 libtool: compile:  cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_statement.c -MMD -MF oci8_statement.dep -MT oci8_statement.lo  -fPIC -DPIC -o .libs/oci8_statement.o
32.08 /bin/bash /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/libtool --tag=CC --mode=compile cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_collection.c -o oci8_collection.lo  -MMD -MF oci8_collection.dep -MT oci8_collection.lo
32.10 libtool: compile:  cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_collection.c -MMD -MF oci8_collection.dep -MT oci8_collection.lo  -fPIC -DPIC -o .libs/oci8_collection.o
32.33 /bin/bash /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/libtool --tag=CC --mode=compile cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_interface.c -o oci8_interface.lo  -MMD -MF oci8_interface.dep -MT oci8_interface.lo
32.35 libtool: compile:  cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_interface.c -MMD -MF oci8_interface.dep -MT oci8_interface.lo  -fPIC -DPIC -o .libs/oci8_interface.o
32.91 /bin/bash /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/libtool --tag=CC --mode=compile cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_failover.c -o oci8_failover.lo  -MMD -MF oci8_failover.dep -MT oci8_failover.lo
32.93 libtool: compile:  cc -I. -I/tmp/pear/temp/oci8 -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include -DHAVE_CONFIG_H -g -O2 -D_GNU_SOURCE -DZEND_COMPILE_DL_EXT=1 -c /tmp/pear/temp/oci8/oci8_failover.c -MMD -MF oci8_failover.dep -MT oci8_failover.lo  -fPIC -DPIC -o .libs/oci8_failover.o
33.02 /bin/bash /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/libtool --tag=CC --mode=link cc -shared -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/include -I/tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/main -I/tmp/pear/temp/oci8 -I/usr/include/php/20220829 -I/usr/include/php/20220829/main -I/usr/include/php/20220829/TSRM -I/usr/include/php/20220829/Zend -I/usr/include/php/20220829/ext -I/usr/include/php/20220829/ext/date/lib -I/opt/oracle/instantclient_18_3//sdk/include  -DHAVE_CONFIG_H  -g -O2 -D_GNU_SOURCE    -o oci8.la -export-dynamic -avoid-version -prefer-pic -module -rpath /tmp/pear/temp/pear-build-defaultuserW5s3lG/oci8-3.3.0/modules  oci8.lo oci8_lob.lo oci8_statement.lo oci8_collection.lo oci8_interface.lo oci8_failover.lo -Wl,-rpath,/opt/oracle/instantclient_18_3/ -L/opt/oracle/instantclient_18_3/ -lclntsh
33.05 libtool: link: cc -shared  -fPIC -DPIC  .libs/oci8.o .libs/oci8_lob.o .libs/oci8_statement.o .libs/oci8_collection.o .libs/oci8_interface.o .libs/oci8_failover.o   -L/opt/oracle/instantclient_18_3/ -lclntsh  -g -O2 -Wl,-rpath -Wl,/opt/oracle/instantclient_18_3/   -Wl,-soname -Wl,oci8.so -o .libs/oci8.so
33.06 /usr/bin/ld: skipping incompatible /opt/oracle/instantclient_18_3//libclntsh.so when searching for -lclntsh
33.06 /usr/bin/ld: cannot find -lclntsh
33.06 collect2: error: ld returned 1 exit status
33.06 make: *** [Makefile:232: oci8.la] Error 1
33.06 ERROR: `make' failed
------
failed to solve: process "/bin/sh -c if [ ${INSTALL_OCI8} = true ]; then   apt-get update && apt-get install --no-install-recommends -y wget   && mkdir /opt/oracle       && cd /opt/oracle       && wget ${ORACLE_INSTANT_CLIENT_MIRROR}instantclient-basic-linux.${ORACLE_INSTANT_CLIENT_ARCH}-${ORACLE_INSTANT_CLIENT_VERSION}.zip       && wget ${ORACLE_INSTANT_CLIENT_MIRROR}instantclient-sdk-linux.${ORACLE_INSTANT_CLIENT_ARCH}-${ORACLE_INSTANT_CLIENT_VERSION}.zip       && unzip /opt/oracle/instantclient-basic-linux.${ORACLE_INSTANT_CLIENT_ARCH}-${ORACLE_INSTANT_CLIENT_VERSION}.zip -d /opt/oracle       && unzip /opt/oracle/instantclient-sdk-linux.${ORACLE_INSTANT_CLIENT_ARCH}-${ORACLE_INSTANT_CLIENT_VERSION}.zip -d /opt/oracle       && if [ ${OCI_VERSION} -lt 18 ] ; then ln -s /opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/libclntsh.so.${ORACLE_INSTANT_CLIENT_MAJOR}.${ORACLE_INSTANT_CLIENT_MINOR} /opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/libclntsh.so ; fi      && if [ ${OCI_VERSION} -lt 18 ] ; then ln -s /opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/libclntshcore.so.${ORACLE_INSTANT_CLIENT_MAJOR}.${ORACLE_INSTANT_CLIENT_MINOR} /opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/libclntshcore.so ; fi       && if [ ${OCI_VERSION} -lt 18 ] ; then ln -s /opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/libocci.so.${ORACLE_INSTANT_CLIENT_MAJOR}.${ORACLE_INSTANT_CLIENT_MINOR} /opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/libocci.so ; fi       && rm -rf /opt/oracle/*.zip   && apt-get update       && apt-get install --no-install-recommends -y           libaio-dev &&   if [ $(php -r \"echo PHP_MAJOR_VERSION;\") = \"5\" ]; then   echo 'instantclient,/opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/' | pecl install oci8-2.0.12;   elif [ $(php -r \"echo PHP_MAJOR_VERSION;\") = \"7\" ]; then   echo 'instantclient,/opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/' | pecl install oci8-2.2.0;   elif [ $(php -r \"echo PHP_VERSION_ID - PHP_RELEASE_VERSION;\") = \"80000\" ]; then     echo \"instantclient,/opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/\" | pecl install oci8-3.0.1;   elif [ $(php -r \"echo PHP_MAJOR_VERSION . PHP_MINOR_VERSION;\") = \"81\" ]; then     echo \"instantclient,/opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/\" | pecl install oci8-3.2.1;   else     echo \"instantclient,/opt/oracle/instantclient_${ORACLE_INSTANT_CLIENT_VERSION}/\" | pecl install oci8;   fi   && echo \"extension=oci8.so\" >> /etc/php/${LARADOCK_PHP_VERSION}/cli/php.ini   && php -m | grep -q 'oci8' ;fi" did not complete successfully: exit code: 1
```

**Comment by parallels999** [comment]
did you try changing?
```sh
ARG ORACLE_INSTANT_CLIENT_MAJOR=18
ARG ORACLE_INSTANT_CLIENT_MINOR=3
```

**Comment by chieroz** [comment]
Hello, and thank for helping. 

The values you suggest are already in the .env file. I tried with the values of the most recent Oracle Instant Client (see below) but as it seems nothing changed. Maybe the ORACLE_INSTANT_CLIENT_MIRROR is obsolete?

```
ARG ORACLE_INSTANT_CLIENT_MAJOR=24
ARG ORACLE_INSTANT_CLIENT_MINOR=1
```

**Comment by parallels999** [comment]
>```
>ARG ORACLE_INSTANT_CLIENT_MAJOR=24
>ARG ORACLE_INSTANT_CLIENT_MINOR=1
>```

What?? why? Where did you read that you should put that version?

Try
```
ORACLE_INSTANT_CLIENT_MAJOR=19
ORACLE_INSTANT_CLIENT_MINOR=10
```

**Comment by chieroz** [comment]
I would be very happy to help the community, but as you see I have big troubles even when I try to help myself...

Thank you for taking some time to help me. I tried with the value you suggest but with no success. Looking at the log I think it's still loading the instantclient-basic-linux.x86_64-18_3.zip file.

24.1 is the version I installed on a Ubuntu 22.04 server following the instructions on the Oracle website.

Thank you VERY much for your help.
